File: sol/1915/1915.py
class Solution:
    def wonderfulSubstrings(self, word: str) -> int:
        n = len(word)
        currMask = 0
        knownMaskCount = [0] * (1 << 10)
        knownMaskCount[0] = 1
        msk = {chr(ord('a') + i) : 1 << i for i in range(10)}
        for c in word:
            currMask ^= msk[c]
            knownMaskCount[currMask] += 1
        res = 0
        for m, c in enumerate(knownMaskCount):
            res += c * (c - 1)
            for i in range(10):
                res += c * knownMaskCount[m ^ (1 << i)]
        return res // 2

File: sol/1915/1915.cpp
struct Solution {
    inline long long wonderfulSubstrings(string &word) {
        int currMask = 0;
        unsigned long long knownMaskCount[1024];
        knownMaskCount[0] = 1;
        for (int i = 1; i < 1024; ++i){
            knownMaskCount[i] = 0;
        }
        for (char &c : word){
            currMask ^= (1 << (c - 'a'));
            ++knownMaskCount[currMask];
        }
        unsigned long long res = 0;
        for (int i = 0; i < 1024; ++i){
            res += knownMaskCount[i] * (knownMaskCount[i] - 1);
            for (int j = 0; j < 10; ++j){
                res += knownMaskCount[i] * knownMaskCount[i ^ (1 << j)];
            }
        }
        return res / 2;
    }
};
